backend/siarnaq/api/episodes/views.py

```diff
@@ -84,6 +86,16 @@ def get_queryset(self):
             queryset = queryset.filter(is_public=True)
         return queryset
 
+    @action(detail=False, methods=["get"])
+    def next(self, request, episode_id):
+        "Retrieve the next upcoming tournament, as ordered by submission freeze time."
+        now = timezone.now()
+        next_tournament = get_object_or_404(
+            self.get_queryset().filter(submission_freeze__gte=now)
```

Comment:
This will crash if there is >1 tournament in the future, because `get_object_or_404` does not work with querysets of length ≥ 2. Use a slice to limit the size of the returned queryset. Note that line 80 (`.order_by("submission_freeze")`) guarantees that the first one is indeed the chronologically next one, and not an arbitrary future tournament.

```suggestion
            self.get_queryset().filter(submission_freeze__gt=now)[:1]
```